src/node/rpc/frontend.h

```diff
@@ -129,7 +134,7 @@ namespace ccf
 
     std::optional<CallerId> valid_caller(Store::Tx& tx, const CBuffer& caller)
     {
-      if (certs == nullptr)
+      if (certs == nullptr || !requires_valid_caller)
```

Comment:
As discussed with @eddyashton and @olgavrou, it would be better to let the `NodeFrontend` override `valid_caller` to apply its specific more permissive logic.